Jenkins release build pipeline fails at integration tests

## Problem:
The Jenkins release build pipeline `release-pipeline.jenkins` fails with these errors at the integration test stage:
```
ProjectMockConfigurationScenario2IntTest > a_user_can_change_mock_configuration_and_behavior FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at ProjectMockConfigurationScenario2IntTest.java:103

SecHubExecutionScenarioSecHubClientIntTest > sechub_client_is_able_to_handle_synchronous_and_result_has_trafficlight_green FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:453

SecHubExecutionScenarioSecHubClientIntTest > a_project_having_no_metadata_but_no_problems_can_be_executed_as_codescan_and_results_green FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:139

SecHubExecutionScenarioSecHubClientIntTest > sechub_client_is_able_to_handle_synchronous_and_result_has_trafficlight_yellow_stop_on_yellow_active_so_exit_code1 FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:505

SecHubExecutionScenarioSecHubClientIntTest > sechub_client_is_able_to_handle_synchronous_and_result_has_trafficlight_red FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:531

SecHubExecutionScenarioSecHubClientIntTest > sechub_client_is_able_to_handle_asynchronous_and_file_is_uploaded FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:399

SecHubExecutionScenarioSecHubClientIntTest > a_project_having_metadata_no_problems_can_be_executed_as_codescan_and_results_green FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:180

SecHubExecutionScenarioSecHubClientIntTest > a_project_having_no_white_list_entries_but_no_problems_can_be_executed_as_codescan_and_results_green FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:103

SecHubExecutionScenarioSecHubClientIntTest > sechub_client_is_able_to_trigger_sourcescan_asynchronous_even_when_userid_is_uppercased FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:54

SecHubExecutionScenarioSecHubClientIntTest > a_project_having_no_problems_can_be_executed_as_codescan_and_results_green FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:281

SecHubExecutionScenarioSecHubClientIntTest > sechub_client_is_able_to_handle_asynchronous_and_file_is_uploaded_and_excludes_handled FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:430

SecHubExecutionScenarioSecHubClientIntTest > a_project_having_no_white_list_entries_but_some_problems_can_be_executed_as_codescan_and_results_yellow FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:367

SecHubExecutionScenarioSecHubClientIntTest > sechub_client_can_execute_a_config_file_which_uses_template_variables_of_environment_entries_but_no_data_section FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:323

SecHubExecutionScenarioSecHubClientIntTest > sechub_client_is_able_to_handle_synchronous_and_result_has_trafficlight_yellow_pe_default_exitcode0 FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:479

SecHubExecutionScenarioSecHubClientIntTest > sechub_client_is_able_to_trigger_infrascan_asynchronous FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:77

SecHubExecutionScenarioSecHubClientIntTest > a_project_having_metadata_updated_no_problems_can_be_executed_as_codescan_and_results_green FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at SecHubExecutionScenarioSecHubClientIntTest.java:244

FalsePositivesScenario3IntTest > REST_API_direct_mark_20_false_positives_with_comments_is_accepted FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at FalsePositivesScenario3IntTest.java:103

FalsePositivesScenario3IntTest > with_sechubclient_mark_falsepositives_of_only_existing_medium_will_result_in_report_without_defined__And_trafficlight_changes_from_yellow_to_green FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at FalsePositivesScenario3IntTest.java:41

FalsePositivesScenario3IntTest > with_sechubclient_unmark_falsepositives_of_only_existing_medium_will_result_in_report_without_defined__And_trafficlight_changes_from_gren_to_yellow FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at FalsePositivesScenario3IntTest.java:173

FalsePositivesScenario3IntTest > with_sechubclient_fetch_fp_config_when_one_entry_added FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at FalsePositivesScenario3IntTest.java:238

FalsePositivesScenario3IntTest > REST_API_direct_fetch_fp_config_when_one_entry_added FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at FalsePositivesScenario3IntTest.java:209

FalsePositivesScenario3IntTest > REST_API_direct_mark_falsepositives_of_only_existing_medium_will_result_in_report_without_defined__And_trafficlight_changes_from_yellow_to_green FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at FalsePositivesScenario3IntTest.java:72

FalsePositivesScenario3IntTest > REST_API_direct_unmark_falsepositives_of_only_existing_medium_will_result_in_report_without_defined__And_trafficlight_changes_from_gren_to_yellow FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at FalsePositivesScenario3IntTest.java:136

FullScanDataScenario3SecHubClientIntTest > product_failure_results_in_downloadable_scan_log FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at FullScanDataScenario3SecHubClientIntTest.java:52

FullScanDataScenario3SecHubClientIntTest > user_1_starts_job_but_only_admin_can_download_scanlog_or_fullscan_data FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at FullScanDataScenario3SecHubClientIntTest.java:86

MappingScenario3IntTest > checkmarx_executor_config_mapping_for_checkmarx_preset_changed_is_used_in_next_checkmarx_scan FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at MappingScenario3IntTest.java:59

ProjectChangeAccessLevelScenario3IntTest > none__test_delete_removes_former_access_level_settings FAILED
    java.lang.IllegalStateException at ProjectChangeAccessLevelScenario3IntTest.java:68
        Caused by: org.springframework.web.client.HttpClientErrorException$NotFound at ProjectChangeAccessLevelScenario3IntTest.java:72

ProjectChangeAccessLevelScenario3IntTest > get_job_report__existing_job_read_access_level_changing_test_different_access_levels FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at ProjectChangeAccessLevelScenario3IntTest.java:121

ProjectDeleteScenario3IntTest > super_admin_deletes_project__deletes_also_all_scan_and_product_results FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at ProjectDeleteScenario3IntTest.java:79

ProductExecutorConfigurationScenario8IntTest > one_profile_with_two_configs_for_same_product_result_in_scanlog_with_separated_metadata_and_result_entries FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at ProductExecutorConfigurationScenario8IntTest.java:53

ProductExecutorConfigurationScenario8IntTest > two_profiles_with_same_config_inside_for_same_product_result_in_scanlog_with_one_metadata_and_result_entry_for_product FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at ProductExecutorConfigurationScenario8IntTest.java:165

ProductExecutorConfigurationScenario8IntTest > two_profiles_with_two_configs_for_same_product_result_in_scanlog_with_separeted_metadata_and_result_entries FAILED
    com.mercedesbenz.sechub.integrationtest.internal.SecHubClientExecutor$SecHubClientNotFoundException at ProductExecutorConfigurationScenario8IntTest.java:110

237 tests completed, 32 failed
```

Please fix.

--- a/continous-integration-multibranch-pipeline.jenkins
+++ b/continous-integration-multibranch-pipeline.jenkins
@@ -55,7 +55,7 @@ pipeline {
                             // on integration phase again (because nothing has changed, so gradle will cache the results which are ignored ...
                             callGradleWrapper("ensureLocalhostCertificate build generateOpenapi buildDeveloperAdminUI -x :sechub-integrationtest:test -x :sechub-cli:build -Psechub.test.wiremock.https_port=${env.SECHUB_TEST_WIREMOCK_HTTPS_PORT} -Psechub.test.wiremock.http_port=${env.SECHUB_TEST_WIREMOCK_HTTP_PORT}")
                             callGradleWrapper(":sechub-api-java:buildAPIJava")
-                    }
+                        }
                     }
                 }
 

--- a/release-pipeline.jenkins
+++ b/release-pipeline.jenkins
@@ -84,48 +84,46 @@ pipeline {
 
         stage('Integration tests') {
             environment {
-               /* against build race conditions - See TestPortProvider.java */
-               SERVER_PORT= getJenkinsExecutorPort(8243)
+                /* against build race conditions - See TestPortProvider.java */
+                SERVER_PORT= getJenkinsExecutorPort(8243)
+                PDS_PORT= getJenkinsExecutorPort(8543)
 
-               /* setup s3 storage when defined as secrets - otherwise sharedvolume will be used with temp */
-               SECHUB_STORAGE_S3_ACCESSKEY = getStringCredentialsOrUndefined('sechub-storage-s3-accesskey')
-               SECHUB_STORAGE_S3_SECRETKEY = getStringCredentialsOrUndefined('sechub-storage-s3-secretkey')
-               SECHUB_STORAGE_S3_BUCKETNAME = getStringCredentialsOrUndefined('sechub-storage-s3-bucketname')
-               SECHUB_STORAGE_S3_ENDPOINT = getStringCredentialsOrUndefined('sechub-storage-s3-endpoint')
-               
-               // we setup PDS storage with same s3 bucket data
-               SECHUB_PDS_STORAGE_S3_ACCESSKEY = getStringCredentialsOrUndefined('sechub-storage-s3-accesskey')
-               SECHUB_PDS_STORAGE_S3_SECRETKEY = getStringCredentialsOrUndefined('sechub-storage-s3-secretkey')
-               SECHUB_PDS_STORAGE_S3_BUCKETNAME = getStringCredentialsOrUndefined('sechub-storage-s3-bucketname')               
-               SECHUB_PDS_STORAGE_S3_ENDPOINT = getStringCredentialsOrUndefined('sechub-storage-s3-endpoint')
+                /* setup s3 storage when defined as secrets - otherwise sharedvolume will be used with temp */
+                SECHUB_STORAGE_S3_ACCESSKEY = getStringCredentialsOrUndefined('sechub-storage-s3-accesskey')
+                SECHUB_STORAGE_S3_SECRETKEY = getStringCredentialsOrUndefined('sechub-storage-s3-secretkey')
+                SECHUB_STORAGE_S3_BUCKETNAME = getStringCredentialsOrUndefined('sechub-storage-s3-bucketname')
+                SECHUB_STORAGE_S3_ENDPOINT = getStringCredentialsOrUndefined('sechub-storage-s3-endpoint')
 
-               SECHUB_INTEGRATIONTEST_PREPARE_WAIT_MS=100
-               
-               /* avoid race conditions with management port */
-               MANAGEMENT_SERVER_PORT= getJenkinsExecutorPort(10250)
+                // we setup PDS storage with same s3 bucket data
+                SECHUB_PDS_STORAGE_S3_ACCESSKEY = getStringCredentialsOrUndefined('sechub-storage-s3-accesskey')
+                SECHUB_PDS_STORAGE_S3_SECRETKEY = getStringCredentialsOrUndefined('sechub-storage-s3-secretkey')
+                SECHUB_PDS_STORAGE_S3_BUCKETNAME = getStringCredentialsOrUndefined('sechub-storage-s3-bucketname')
+                SECHUB_PDS_STORAGE_S3_ENDPOINT = getStringCredentialsOrUndefined('sechub-storage-s3-endpoint')
 
+                SECHUB_INTEGRATIONTEST_PREPARE_WAIT_MS=100
+
+                /* avoid race conditions with management port */
+                MANAGEMENT_SERVER_PORT= getJenkinsExecutorPort(10250)
             }
-
             steps {
-              script{
-                  callGradleWrapper("integrationtest -Psechub.integrationtest.serverport=${env.SERVER_PORT}")
-               }
+                script{
+                    callGradleWrapper("integrationtest -Psechub.integrationtest.serverport=${env.SERVER_PORT} -Psechub.integrationtest.pdsport=${env.PDS_PORT}")
+                }
             }
         }
 
         stage('Build Documentation') {
             steps {
-              script{
-                  callGradleWrapper('documentation')
-               }
+                script{
+                    callGradleWrapper('documentation')
+                }
             }
         }
 
         stage('Publish release artifacts') {
             when {
-                 expression { context.isAnyRelease() }
+                expression { context.isAnyRelease() }
             }
-            
             steps {
                 echo 'This step is now ignored - because already done by GitHub actions - see https://github.com/mercedes-benz/sechub/actions'
             }
@@ -140,12 +138,11 @@ pipeline {
                  expression { context.isServerRelease() }
             }
             steps {
-                    // trigger other pipeline job :
-                    build job: 'sechub-server-released',
-                       propagate: true,
-                       wait: false,
-                       parameters: [string(name: 'SERVERVERSION', value:"${context.serverVersion}")]
-
+                // trigger other pipeline job :
+                build job: 'sechub-server-released',
+                    propagate: true,
+                    wait: false,
+                    parameters: [string(name: 'SERVERVERSION', value:"${context.serverVersion}")]
             }
         }
 
@@ -154,12 +151,11 @@ pipeline {
                  expression { context.isClientRelease() }
             }
             steps {
-                    // trigger other pipeline job :
-                    build job: 'sechub-client-released',
-                       propagate: true,
-                       wait: false,
-                       parameters: [string(name: 'CLIENTVERSION', value:"${context.clientVersion}")]
-
+                // trigger other pipeline job :
+                build job: 'sechub-client-released',
+                    propagate: true,
+                    wait: false,
+                    parameters: [string(name: 'CLIENTVERSION', value:"${context.clientVersion}")]
             }
         }
 
@@ -168,12 +164,11 @@ pipeline {
                  expression { context.isPdsRelease() }
             }
             steps {
-                    // trigger other pipeline job :
-                    build job: 'sechub-pds-released',
-                       propagate: true,
-                       wait: false,
-                       parameters: [string(name: 'PDS_VERSION', value:"${context.pdsVersion}")]
-
+                // trigger other pipeline job :
+                build job: 'sechub-pds-released',
+                    propagate: true,
+                    wait: false,
+                    parameters: [string(name: 'PDS_VERSION', value:"${context.pdsVersion}")]
             }
         }
 

--- a/release-pipeline.jenkins
+++ b/release-pipeline.jenkins
@@ -40,6 +40,7 @@ pipeline {
                             to: '$DEFAULT_RECIPIENTS'
                         )
                     }
+                    callGradleWrapper("clean")
                 }
             }
         }
@@ -66,7 +67,8 @@ pipeline {
                             * Reason: because we do NOT want to have the integration tests executed, otherwise gradle will not execute them
                             * on integration phase again (because nothing has changed, so gradle will cache the results which are ignored ...
                             */
-                            callGradleWrapper("clean ensureLocalhostCertificate build -x :sechub-integrationtest:test -x :sechub-cli:build -Psechub.test.wiremock.https_port=${env.SECHUB_TEST_WIREMOCK_HTTPS_PORT} -Psechub.test.wiremock.http_port=${env.SECHUB_TEST_WIREMOCK_HTTP_PORT}")
+                            callGradleWrapper("ensureLocalhostCertificate build generateOpenapi buildDeveloperAdminUI -x :sechub-integrationtest:test -x :sechub-cli:build -Psechub.test.wiremock.https_port=${env.SECHUB_TEST_WIREMOCK_HTTPS_PORT} -Psechub.test.wiremock.http_port=${env.SECHUB_TEST_WIREMOCK_HTTP_PORT}")
+                            callGradleWrapper(":sechub-api-java:buildAPIJava")
                         }
                     }
                 }
@@ -107,6 +109,7 @@ pipeline {
             }
             steps {
                 script{
+                    // start integration tests and also create combined test report after all
                     callGradleWrapper("integrationtest -Psechub.integrationtest.serverport=${env.SERVER_PORT} -Psechub.integrationtest.pdsport=${env.PDS_PORT}")
                 }
             }

--- a/release-pipeline.jenkins
+++ b/release-pipeline.jenkins
@@ -40,44 +40,48 @@ pipeline {
                             to: '$DEFAULT_RECIPIENTS'
                         )
                     }
+                }
+            }
+        }
 
+        stage("Builds") {
+            parallel {
+
+                stage('Build Server and PDS') {
+                    environment {
+                        /* against build race conditions - See TestPortProvider.java */
+                        SECHUB_TEST_WIREMOCK_HTTPS_PORT = getJenkinsExecutorPort(8143)
+                        SECHUB_TEST_WIREMOCK_HTTP_PORT = getJenkinsExecutorPort(8180)
+
+                        SECHUB_TEST_S3MOCK_HTTP_PORT = getJenkinsExecutorPort(9090)
+                        SECHUB_TEST_S3MOCK_HTTPS_PORT = getJenkinsExecutorPort(9190)
+                    }
+                    steps {
+                        script{
+                            echo "SECHUB_TEST_WIREMOCK_HTTPS_PORT=${env.SECHUB_TEST_WIREMOCK_HTTPS_PORT}"
+                            echo "SECHUB_TEST_WIREMOCK_HTTP_PORT=${env.SECHUB_TEST_WIREMOCK_HTTP_PORT}"
+
+                            /*
+                            * We do NOT build sechub-integrationtest
+                            * Reason: because we do NOT want to have the integration tests executed, otherwise gradle will not execute them
+                            * on integration phase again (because nothing has changed, so gradle will cache the results which are ignored ...
+                            */
+                            callGradleWrapper("clean ensureLocalhostCertificate build -x :sechub-integrationtest:test -x :sechub-cli:build -Psechub.test.wiremock.https_port=${env.SECHUB_TEST_WIREMOCK_HTTPS_PORT} -Psechub.test.wiremock.http_port=${env.SECHUB_TEST_WIREMOCK_HTTP_PORT}")
+                        }
+                    }
                 }
 
+                stage('Build Client') {
+                    steps {
+                        script{
+                            callGradleWrapper(':sechub-cli:buildGo :sechub-cli:testGo')
+                        }
+                    }
+                }
 
             }
         }
 
-        stage('Build Server') {
-          environment {
-             /* against build race conditions - See TestPortProvider.java */
-             SECHUB_TEST_WIREMOCK_HTTPS_PORT = getJenkinsExecutorPort(8143)
-             SECHUB_TEST_WIREMOCK_HTTP_PORT = getJenkinsExecutorPort(8180)
-
-             SECHUB_TEST_S3MOCK_HTTP_PORT = getJenkinsExecutorPort(9090)
-             SECHUB_TEST_S3MOCK_HTTPS_PORT = getJenkinsExecutorPort(9190)
-          }
-          steps {
-              script{
-                  echo "SECHUB_TEST_WIREMOCK_HTTPS_PORT=${env.SECHUB_TEST_WIREMOCK_HTTPS_PORT}"
-                  echo "SECHUB_TEST_WIREMOCK_HTTP_PORT=${env.SECHUB_TEST_WIREMOCK_HTTP_PORT}"
-
-                  /*
-                   * we do NOT build sechub-integrationtest
-                   * Reason: because we do NOT want to have the integration tests executed, otherwise gradle will not execute them
-                   * on integration phase again (because nothing has changed, so gradle will cache the results which are ignored ...
-                   */
-                  callGradleWrapper("clean ensureLocalhostCertificate build -x :sechub-integrationtest:test -x :sechub-cli:build -Psechub.test.wiremock.https_port=${env.SECHUB_TEST_WIREMOCK_HTTPS_PORT} -Psechub.test.wiremock.http_port=${env.SECHUB_TEST_WIREMOCK_HTTP_PORT}")
-
-               }
-            }
-        }
-        stage('Build CLI') {
-            steps {
-              script{
-                  callGradleWrapper(':sechub-cli:buildGo :sechub-cli:testGo')
-               }
-            }
-        }
         stage('Integration tests') {
             environment {
                /* against build race conditions - See TestPortProvider.java */
@@ -123,12 +127,12 @@ pipeline {
             }
             
             steps {
-                echo 'This step is now ignored - because alreadyd done by GitHub actions - see https://github.com/mercedes-benz/sechub/actions'
+                echo 'This step is now ignored - because already done by GitHub actions - see https://github.com/mercedes-benz/sechub/actions'
             }
         }
 
         /**
-         * How to handle this in your build ?
+         * How to handle these triggers in your build ?
          * See /sechub-doc/src/docs/asciidoc/documents/buildserver/jenkins-pipeline.adoc
          */
         stage('Trigger server released') {
@@ -144,10 +148,7 @@ pipeline {
 
             }
         }
-        /**
-         * How to handle this in your build ?
-         * See /sechub-doc/src/docs/asciidoc/documents/buildserver/jenkins-pipeline.adoc
-         */
+
         stage('Trigger client released') {
             when {
                  expression { context.isClientRelease() }
@@ -162,10 +163,24 @@ pipeline {
             }
         }
 
+        stage('Trigger PDS released') {
+            when {
+                 expression { context.isPdsRelease() }
+            }
+            steps {
+                    // trigger other pipeline job :
+                    build job: 'sechub-pds-released',
+                       propagate: true,
+                       wait: false,
+                       parameters: [string(name: 'PDS_VERSION', value:"${context.pdsVersion}")]
+
+            }
+        }
 
     }
 
     post {
+
         failure {
             script {
                 if (context.isAnyRelease()){
@@ -186,19 +201,21 @@ pipeline {
                 }
             }
         }
+
         success {
-                script {
-                    if (context.isSuccesssEmailNeeded()){
+            script {
+                if (context.isAnyRelease()){
                     releaseInfo = context.getReleaseInfo()
                     emailext (
                         subject:"[ SUCCESS ]Release - Build:${env.BUILD_NUMBER}",
                         body:"Continous integration build with release publish sucessfully done. Triggered additional jobs.\n$releaseInfo\n\n"+
-                             "${env.BUILD_URL}",
+                            "${env.BUILD_URL}",
                         to: '$DEFAULT_RECIPIENTS'
                     )
                 }
-              }
+            }
         }
+
         always {
             junit '**/build/test-results/*/TEST-*.xml'
             script{
@@ -207,6 +224,7 @@ pipeline {
             archive 'build/archive/combined-sechub-testreport.zip'
         }
     }
+
 }
 
 // --------------------------------------------------------------------------
@@ -267,12 +285,15 @@ class SecHubBuildContext{
 
     String clientVersion
     String serverVersion
+    String pdsVersion
 
     boolean clientRelease
     boolean serverRelease
+    boolean pdsRelease
 
-    boolean clientBuildDone
-    boolean serverBuildDone
+    // Obsolete?
+    // boolean clientBuildDone
+    // boolean serverBuildDone
 
     String releaseInfoString
 
@@ -297,29 +318,30 @@ class SecHubBuildContext{
         return result.substring(0,index);
     }
 
-    public boolean isStartEmailNeeded(){
-        return isAnyRelease()
-    }
-
-    public boolean isFailEmailNeeded(){
-        return true;// always
-    }
-
-    public boolean isSuccesssEmailNeeded(){
-        return isAnyRelease()
-    }
-
-    public void setClientBuildDone(boolean done){
-        this.clientBuildDone=done
-    }
-
-    public void setServerBuildDone(boolean done){
-        this.serverBuildDone=done
-    }
-
-    public void setWebsiteBuildDone(boolean done){
-        this.websiteBuildDone=done
-    }
+    // Obsolete?
+    // public boolean isStartEmailNeeded(){
+    //     return isAnyRelease()
+    // }
+    // 
+    // public boolean isFailEmailNeeded(){
+    //     return true; // always
+    // }
+    // 
+    // public boolean isSuccesssEmailNeeded(){
+    //     return isAnyRelease()
+    // }
+    // 
+    // public void setClientBuildDone(boolean done){
+    //     this.clientBuildDone=done
+    // }
+    // 
+    // public void setServerBuildDone(boolean done){
+    //     this.serverBuildDone=done
+    // }
+    // 
+    // public void setWebsiteBuildDone(boolean done){
+    //     this.websiteBuildDone=done
+    // }
 
 
     /**
@@ -330,33 +352,36 @@ class SecHubBuildContext{
      *  - a client deployment should have also integration test - means server build before
      *    to have a working local integration test server
      */
-    public boolean isClientOrServerRelease(){
-        return isClientRelease() || isServerRelease();
-    }
+
+    // Obsolete?
+    // public boolean isClientOrServerRelease(){
+    //     return isClientRelease() || isServerRelease();
+    // }
 
     public boolean isAnyRelease(){
-        return isClientRelease() || isServerRelease();
+        return isClientRelease() || isServerRelease() || isPdsRelease();
     }
 
-    public boolean isSuccessFul(){
-        if (! isAnyRelease()){
-            return false
-        }
-
-        if (isClientRelease()) {
-           if (! this.clientBuildDone){
-               return false
-           }
-        }
-
-        if (isServerRelease()) {
-           if (! this.serverBuildDone){
-               return false
-           }
-        }
-
-        return true
-    }
+    // Obsolete?
+    // public boolean isSuccessFul(){
+    //     if (! isAnyRelease()){
+    //         return false
+    //     }
+    //
+    //     if (isClientRelease()) {
+    //        if (! this.clientBuildDone){
+    //            return false
+    //        }
+    //     }
+    //
+    //     if (isServerRelease()) {
+    //        if (! this.serverBuildDone){
+    //            return false
+    //        }
+    //     }
+    //
+    //     return true
+    // }
 
     public void init(String[ ] commitTags){
         def versionCommitTags = commitTags.findAll {it.startsWith("v")}
@@ -364,6 +389,7 @@ class SecHubBuildContext{
         // we got vX.Y.Z-server and vX.Y.Z-client tags
         def clientVersionCommitTag = versionCommitTags.find{ it.contains("-client") }
         def serverVersionCommitTag = versionCommitTags.find{ it.contains("-server") }
+        def pdsVersionCommitTag = versionCommitTags.find{ it.contains("-pds") }
 
         releaseInfoString = "Release(s):"
         if (clientVersionCommitTag == null || clientVersionCommitTag.empty){
@@ -381,12 +407,18 @@ class SecHubBuildContext{
             serverVersion=getSimplifiedVersion(serverVersionCommitTag)
             releaseInfoString=releaseInfoString+" " + serverVersionCommitTag
         }
+
+        if (pdsVersionCommitTag == null || pdsVersionCommitTag.empty){
+            pdsRelease=false;
+        }else{
+            pdsRelease=true;
+            pdsVersion=getSimplifiedVersion(pdsVersionCommitTag)
+            releaseInfoString=releaseInfoString+" " + pdsVersionCommitTag
+        }
     }
 
     public String getReleaseInfo(){
         return releaseInfoString
     }
 
-
 }
-

--- a/release-pipeline.jenkins
+++ b/release-pipeline.jenkins
@@ -1,8 +1,8 @@
 // SPDX-License-Identifier: MIT
 
 // Release  pipeline script for jenkins
-// Only master branch is  build by this script - other branches are build by "continous-integration-multibranch-pipeline.jenkins"
-// For details please refer /sechub-doc/src/docs/asciidoc/documents/buildserver/jenkins-pipeline.adoc
+// Only the master branch is built by this script - the other branches are built by "continous-integration-multibranch-pipeline.jenkins"
+// For details please refer to /sechub-doc/src/docs/asciidoc/documents/buildserver/jenkins-pipeline.adoc
 
 def context = new SecHubBuildContext();
 def sechubGitBranch ="unknown"
@@ -289,10 +289,6 @@ class SecHubBuildContext{
     boolean serverRelease
     boolean pdsRelease
 
-    // Obsolete?
-    // boolean clientBuildDone
-    // boolean serverBuildDone
-
     String releaseInfoString
 
     public SecHubBuildContext(){
@@ -316,70 +312,9 @@ class SecHubBuildContext{
         return result.substring(0,index);
     }
 
-    // Obsolete?
-    // public boolean isStartEmailNeeded(){
-    //     return isAnyRelease()
-    // }
-    // 
-    // public boolean isFailEmailNeeded(){
-    //     return true; // always
-    // }
-    // 
-    // public boolean isSuccesssEmailNeeded(){
-    //     return isAnyRelease()
-    // }
-    // 
-    // public void setClientBuildDone(boolean done){
-    //     this.clientBuildDone=done
-    // }
-    // 
-    // public void setServerBuildDone(boolean done){
-    //     this.serverBuildDone=done
-    // }
-    // 
-    // public void setWebsiteBuildDone(boolean done){
-    //     this.websiteBuildDone=done
-    // }
-
-
-    /**
-     * Returns true when client or sever deployment.
-     * Interesting for interdepent deployment tests.
-     * E.g.
-     *  - a server deployment should have integration test + client build before.
-     *  - a client deployment should have also integration test - means server build before
-     *    to have a working local integration test server
-     */
-
-    // Obsolete?
-    // public boolean isClientOrServerRelease(){
-    //     return isClientRelease() || isServerRelease();
-    // }
-
     public boolean isAnyRelease(){
         return isClientRelease() || isServerRelease() || isPdsRelease();
     }
-
-    // Obsolete?
-    // public boolean isSuccessFul(){
-    //     if (! isAnyRelease()){
-    //         return false
-    //     }
-    //
-    //     if (isClientRelease()) {
-    //        if (! this.clientBuildDone){
-    //            return false
-    //        }
-    //     }
-    //
-    //     if (isServerRelease()) {
-    //        if (! this.serverBuildDone){
-    //            return false
-    //        }
-    //     }
-    //
-    //     return true
-    // }
 
     public void init(String[ ] commitTags){
         def versionCommitTags = commitTags.findAll {it.startsWith("v")}

--- a/release-pipeline.jenkins
+++ b/release-pipeline.jenkins
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: MIT
 
-// Release  pipeline script for jenkins
+// Release pipeline script for jenkins
 // Only the master branch is built by this script - the other branches are built by "continous-integration-multibranch-pipeline.jenkins"
 // For details please refer to /sechub-doc/src/docs/asciidoc/documents/buildserver/jenkins-pipeline.adoc
 
@@ -46,6 +46,9 @@ pipeline {
         }
 
         stage("Builds") {
+            when {
+                expression { context.isAnyRelease() }
+            }
             parallel {
 
                 stage('Build Server and PDS') {
@@ -85,6 +88,9 @@ pipeline {
         }
 
         stage('Integration tests') {
+            when {
+                expression { context.isAnyRelease() }
+            }
             environment {
                 /* against build race conditions - See TestPortProvider.java */
                 SERVER_PORT= getJenkinsExecutorPort(8243)
@@ -116,6 +122,9 @@ pipeline {
         }
 
         stage('Build Documentation') {
+            when {
+                expression { context.isAnyRelease() }
+            }
             steps {
                 script{
                     callGradleWrapper('documentation')

